lib/keila_web/controllers/api_controller.ex

```diff
@@ -0,0 +1,231 @@
+defmodule KeilaWeb.ApiController do
+  use KeilaWeb, :controller
+
+  alias Keila.Auth
+  alias Keila.Auth.Token
+  alias Keila.Projects
+  alias KeilaWeb.ApiNormalizer
+  alias Keila.Contacts
+  alias Keila.Mailings
+
+  plug :authorize
+
+  #
+  # Contact functions
+  #
+
+  plug ApiNormalizer, [normalize: [:pagination, :contacts_filter]] when action == :index_contacts
+
+  plug ApiNormalizer,
+       [normalize: [{:data, :contact}]] when action in [:create_contact, :update_contact]
+
+  @spec index_contacts(Plug.Conn.t(), Map.t()) :: Plug.Conn.t()
+  def index_contacts(conn, _params) do
+    contacts =
+      Contacts.get_project_contacts(project_id(conn),
+        paginate: conn.assigns.pagination,
+        filter: conn.assigns.filter
+      )
+
+    render(conn, "contacts.json", %{contacts: contacts})
+  end
+
+  def create_contact(conn, _params) do
+    case Contacts.create_contact(project_id(conn), conn.assigns.data) do
+      {:ok, contact} -> render(conn, "contact.json", %{contact: contact})
+      {:error, changeset} -> send_changeset_error(conn, changeset)
+    end
+  end
+
+  def show_contact(conn, %{"id" => id}) do
+    case Contacts.get_project_contact(project_id(conn), id) do
+      contact = %Contacts.Contact{} -> render(conn, "contact.json", %{contact: contact})
+      nil -> send_404(conn)
+    end
+  end
+
+  def update_contact(conn, %{"id" => id}) do
+    if Contacts.get_project_contact(project_id(conn), id) do
+      case Contacts.update_contact(id, conn.assigns.data) do
+        {:ok, contact} -> render(conn, "contact.json", %{contact: contact})
+        {:error, changeset} -> send_changeset_error(conn, changeset)
+      end
+    else
+      send_404(conn)
+    end
+  end
+
+  def delete_contact(conn, %{"id" => id}) do
+    Contacts.delete_project_contacts(project_id(conn), filter: %{"id" => id})
+
+    conn
+    |> put_status(204)
+  end
+
+  #
+  # Campaign functions
+  #
+
+  plug ApiNormalizer,
+       [normalize: [{:data, :campaign}]]
```

Comment:
I don't see the implementation of :campaign and :segment normalizer, is this correct?